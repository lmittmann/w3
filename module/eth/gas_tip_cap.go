package eth

import (
	"math/big"

	"github.com/lmittmann/w3/internal/module"
	"github.com/lmittmann/w3/w3types"
)

// GasTipCap requests the currently suggested gas tip cap after EIP-1559 to
// allow a timely execution of a transaction.
func GasTipCap() w3types.RPCCallerFactory[*big.Int] {
	return module.NewFactory(
		"eth_maxPriorityFeePerGas",
		nil,
		module.WithRetWrapper(module.HexBigRetWrapper),
	)
}
