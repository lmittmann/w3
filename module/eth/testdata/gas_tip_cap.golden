> {"jsonrpc":"2.0","id":1,"method":"eth_maxPriorityFeePerGas"}
< {"jsonrpc":"2.0","id":1,"result":"0xc0fe"}
